### PR TITLE
KAFKA-3020: Ensure CheckStyle runs on all Java code

### PR DESCRIPTION
**Pull request by granthenke**
- Adds CheckStyle to core and examples modules
- Fixes any existing CheckStyle issues

### CODE REVIEW

File: checkstyle/import-control-core.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  <!-- THINK HARD ABOUT THE LAYERING OF THE PROJECT BEFORE CHANGING THIS FILE -->
+
+  <!-- common library dependencies -->
+  <allow pkg="java" />
+  <allow pkg="scala" />
+  <allow pkg="javax.management" />
+  <allow pkg="org.slf4j" />
+  <allow pkg="org.junit" />
+  <allow pkg="org.easymock" />
+  <allow pkg="org.powermock" />
```
Comment (ewencp) [inline checkstyle/import-control-core.xml:32]: Doesn't look like powermock is used in core?

File: checkstyle/import-control-core.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  <allow pkg="java.security" />
+  <allow pkg="javax.net.ssl" />
+  <allow pkg="javax.security" />
+  <allow pkg="org.ietf.jgss" />
```
Comment (ewencp) [inline checkstyle/import-control-core.xml:36]: Doesn't look like this is used in core anywhere?

File: core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java [hunk 3]
```diff
@@ -17,72 +17,72 @@
...
-   *  @param valueDecoder a decoder that decodes the message itself
-   *  @return a map of (topic, list of  KafkaStream) pairs.
-   *          The number of items in the list is #streams. Each stream supports
-   *          an iterator over message/metadata pairs.
-   */
-  public <K,V> Map<String, List<KafkaStream<K,V>>> 
-    createMessageStreams(Map<String, Integer> topicCountMap, Decoder<K> keyDecoder, Decoder<V> valueDecoder);
-  
-  public Map<String, List<KafkaStream<byte[], byte[]>>> createMessageStreams(Map<String, Integer> topicCountMap);
+    /**
```
Comment (ewencp) [inline core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java:30]: Seems like a lot of the changes in these files are just re-spacing? Not sure if we want to do this, although it is a pain that they are 2-space indented matching the Scala code despite being java files. In particular, there are at least a couple where the imports haven't even changed (which is why I'd expect them to make it into this patch).

Comment (granthenke) [inline core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java:30]: I will change back to 2 space.

Comment (granthenke) [inline core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java:30]: @ewencp Actually looking closer (I made these changes a bit ago) it looks like we have rules to check indentation level for java code.

Comment (ewencp) [inline core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java:30]: @granthenke Ahh, great point! I guess that is both nice to enforce and also annoying in this particular case.... Seems like we did some significant reformatting under the initial checkstyle addition as well, which unfortunately messes up git-blame, but I guess it's a one-time cost.

Comment (granthenke) [inline core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java:30]: @ewencp If there is someway to remove any credit/blame I don't really care about the "credit". So feel free.

Good news is this code wont likely live too much longer and is just for old clients and examples.

Comment (ewencp) [inline core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java:30]: @granthenke Heh, I didn't even mean for the actual credit. But sometimes it's useful in tracking down where an issue was first introduced, the reasoning for the way a particular chunk of code is written, etc. It's an inconvenience, not a serious problem, especially for a set of changes this size.

Comment (granthenke) [inline core/src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java:30]: @ewencp In that case the [-w option](https://git-scm.com/docs/git-blame) on the blame command should work.

Ex: `git blame -w --minimal src/main/scala/kafka/javaapi/consumer/ConsumerConnector.java`

**Comment by ewencp** [comment]
@granthenke LGTM, merged with minor tweak to remove unused imports in the import-control-core.xml.

**Comment by granthenke** [comment]
@ewencp Thanks!

**Pull request closed without merging**